src/System.Globalization.Extensions/tests/IdnMapping/Data/Unicode_6_0/Unicode_6_0_IdnaTest.cs

```diff
@@ -69,7 +68,10 @@ private static string EscapedToLiteralString(string escaped, int lineNumber)
                 if (i + 1 < escaped.Length && escaped[i] == '\\' && escaped[i + 1] == 'u')
                 {
                     // Verify that the escaped sequence is not malformed
-                    Assert.True(i + 5 < escaped.Length, "There was a problem converting to literal string on Line " + lineNumber);
+                    if (i + 5 >= escaped.Length)
+                    {
+                        throw new ArgumentException($"There was a problem converting to literal string on Line {lineNumber}");
+                    }
```

Comment:
I am still seeing it is better to keep the assert as the message will be clear if something goes wrong with the test source data. having the test throw exception like that will looks strange in the test reports comparing with the Assert.